pkg/build/buildutil/util.go

```diff
@@ -257,8 +272,36 @@ func GetBuildSystemConfigMapName(build *buildv1.Build) string {
 // label in a pod. If the length of the string parameter exceeds
 // the maximum label length, the value will be truncated.
 func LabelValue(name string) string {
-	if len(name) <= validation.DNS1123LabelMaxLength {
-		return name
+	end := validation.DNS1123LabelMaxLength
+	newName := name
+	if len(newName) > end {
+		newName = newName[:end]
+	}
+	// first, try to truncate from the end to find a valid
+	// label
+	for end > 0 {
+		if labelRegexpCompile.MatchString(newName) {
+			return newName
+		}
+		end--
+		newName = newName[:end]
+	}
+	// otherwise, try to truncate from the both ends
```

Comment:
I'm inclined to not even deal with this flavor, since this name is suppose to be derived from the build config name, which has already gone through similar validations.

But I leave it up for now to allow for comments / opinions from others.